invenio_iiif/__init__.py

```diff
@@ -6,7 +6,141 @@
 # Invenio is free software; you can redistribute it and/or modify it
 # under the terms of the MIT License; see LICENSE file for more details.
 
-"""IIIF API for Invenio."""
+"""Invenio module to serve images complying with IIIF standard.
+
+Invenio-IIIF integrates
+`Invenio-Records-Files <https://invenio-records-files.rtfd.io>`_ with
+`Flask-IIIF <https://flask-iiif.rtfd.io>`_ to provide an endpoint for serving
+images complying with the `International Image Interoperability Framework
+(IIIF) <https://iiif.io/>`_ API standards.
+
+Invenio-IIIF registers the REST API endpoint provided by Flask-IIIF in the
+Invenio instance through entry points. On each request, it delegates
+authorization check to
+`Invenio-Files-REST <https://invenio-files-rest.rtfd.io>`_, load the file from
+the Invenio bucket and serve it after Flask-IIIF manipulation.
+
+Initialization
+--------------
+
+First create a Flask application (Flask-CLI is not needed for Flask
+version 1.0+):
+
+>>> from flask import Flask
+>>> app = Flask('myapp')
+>>> app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite://'
+
+Configuration
+~~~~~~~~~~~~~
+
+Then, You can define the IIIF API prefix by setting the Invenio configuration
+``IIIF_UI_URL``.
+
+>>> app.config.update(IIIF_UI_URL='/iiif-demo')
+
+Usage
+-----
+
+To demonstrate how Invenio-IIIF works, we need to have a pdf or image file
+uploaded to Invenio instance. To achieve that we need to initialize all
+dependent Invenio extensions:
+
+>>> from invenio_db import InvenioDB, db
+>>> from invenio_files_rest import InvenioFilesREST
+>>> ext_db = InvenioDB(app)
+>>> ext_files_rest = InvenioFilesREST(app)
+>>> app.app_context().push()
+>>> db.create_all()
+
+Creating a location
+~~~~~~~~~~~~~~~~~~~
+
+Before uploading a file we need to create a location:
+
+>>> import tempfile
+>>> from invenio_files_rest.models import Bucket
+>>> tmpdir = tempfile.mkdtemp()
+>>> from invenio_files_rest.models import Location
+>>> loc = Location(name='local', uri=tmpdir, default=True)
+>>> db.session.add(loc)
+>>> db.session.commit()
+
+Creating a bucket with a previously created location:
```

Comment:
Shouldn't it be within instead of with?
`Creating a bucket within the previously created location`
Source https://github.com/inveniosoftware/invenio-files-rest/blob/master/invenio_files_rest/models.py#L25